packages/vite/scripts/dev.ts

```diff
@@ -0,0 +1,108 @@
+import { mkdirSync, readFileSync, rmSync, writeFileSync } from 'node:fs'
+import { type BuildOptions, context } from 'esbuild'
+import packageJSON from '../package.json'
+
+rmSync('dist', { force: true, recursive: true })
+mkdirSync('dist')
+mkdirSync('dist/node')
+writeFileSync('dist/node/index.d.ts', "export * from '../../src/node/index.ts'")
+writeFileSync(
+  'dist/node/runtime.d.ts',
+  "export * from '../../src/runtime/index.ts'",
+)
+
+const serverOptions: BuildOptions = {
+  bundle: true,
+  platform: 'node',
+  target: 'node18',
+  sourcemap: true,
+  external: [
+    ...Object.keys(packageJSON.dependencies),
+    ...Object.keys(packageJSON.peerDependencies),
+    ...Object.keys(packageJSON.optionalDependencies),
+    ...Object.keys(packageJSON.devDependencies),
+  ],
+}
+const clientOptions: BuildOptions = {
+  bundle: true,
+  platform: 'browser',
+  target: 'safari14',
```

Comment:
Should this be es2019 instead as `src/client/tsconfig.json` has `"target": "ES2019"` too? (Not sure if esbuild picks this up automatically too)

Reply:
hum yeah you're write maybe esbuild can pick this up, I'll check!

Reply:
Actually esbuild only use [target to downgrade with class fields](https://esbuild.github.io/content-types/#tsconfig-json:~:text=Note%20that%20the%20target%20setting%20in%20tsconfig.json%20is%20only%20used%20by%20esbuild%20for%20determining%20the%20default%20value%20of%20useDefineForClassFields.%20It%20does%20not%20affect%20esbuild%27s%20own%20target%20setting%2C%20even%20though%20they%20have%20the%20same%20name.)

I updated to es2019 so it should match the previous behaviour (even if in fact the git diff showed nothing in the current client code)

Reply:
I changed to `es2020` I was getting a warning that we use `import.meta` which is part of `es2020`